test/powershell/Modules/Microsoft.PowerShell.Utility/Sort-Object.Tests.ps1

```diff
@@ -222,3 +222,229 @@ Describe "Sort-Object DRT Unit Tests" -Tags "CI" {
 		$results[2] | Should Be 1
 	}
 }
+
+Describe 'Sort-Object Top and Bottom Unit Tests' -Tags 'CI' {
+
+	# Helper function to compare two sort entries
+	function Compare-SortEntry
+	{
+		param($nSortEntry, $fullSortEntry)
+		if ($nSortEntry -is [System.Array]) {
+			# Arrays must be compared differently than everything else
+			$nSortEntry.Equals($fullSortEntry) | Should Be $true
+		} else {
+			$nSortEntry | Should Be $fullSortEntry
+		}
+	}
+
+	# Helper function that compares a full sort with an n-sort
+	function Test-SortObject {
+		param([array]$unsortedData, [hashtable]$baseSortParameters, [string]$nSortType, [int]$nValue)
+		$nSortParameters = @{
+			$nSortType = $nValue
+		}
+		# Sort the data
+		$fullSortResults = $unsortedData | Sort-Object @baseSortParameters
+		$nSortResults = $unsortedData | Sort-Object @baseSortParameters @nSortParameters
+		# Verify the counts when not doing a -Unique sort
+		if (-not $baseSortParameters.ContainsKey('Unique')) {
+			$nSortResults.Count | Should Be $(if ($nSortParameters[$nSortType] -gt $unsortedData.Length) {$unsortedData.Length} else {$nSortParameters[$nSortType]})
+			$fullSortResults.Count | Should Be $unsortedData.Length
+		}
+		# Compare the n-sort result entries with their corresponding full sort result entries
+		if ($nSortType -eq 'Top') {
+			$range = 0..$($nSortResults.Count - 1)
+		} else {
+			$range = -$nSortResults.Count..-1
+		}
+		foreach ($i in $range) {
+			Compare-SortEntry $nSortResults[$i] $fullSortResults[$i]
+		}
+	}
+
+	# Test cases when only the n-sort type needs to be changed
+	$topBottom = @(
+		@{nSortType='Top'   }
+		@{nSortType='Bottom'}
+	)
```

Comment:
> it does not guarantee the order of execution
Where are you getting that from? Pester is not documented that way, and based on how PowerShell works I don't believe that's the case. And if it is changed to work that way, then plenty of Pester tests could start breaking.

You've shared one example where BeforeAll is being used to define functions, etc. however there are many other examples where BeforeAll is not being used in the test harness for PowerShell Core yet variables are assigned, functions are loaded, etc. There's nothing in the Pester documentation about it running tests asynchronously, and even if it did, it would still have to load those tests, which would execute the code for those tests. Changing that behaviour in Pester would be a breaking change.

Since they are not documented, and since I don't have time to go read the code to figure out what they do, I can only ascertain that BeforeAll/AfterAll are much like begin/end blocks, where you can do resource management, make sure certain things are loaded before tests and unloaded afterwards. Other than that, I don't believe there is any advantage whatsoever to using BeforeAll to define the functions/variables that I am defining inside of my Context blocks inside of a BeforeAll block instead.